recipes/corrade/all/cmake/conan-corrade-vars.cmake

```diff
@@ -0,0 +1,61 @@
+
+# Here we are reproducing the variables and call performed by the FindCorrade.cmake provided by the library
+
+# Read flags from configuration
+file(READ "${CMAKE_CURRENT_LIST_DIR}/../../include/Corrade/configure.h" _corradeConfigure)
+string(REGEX REPLACE ";" "\\\\;" _corradeConfigure "${_corradeConfigure}")
+string(REGEX REPLACE "\n" ";" _corradeConfigure "${_corradeConfigure}")
+set(_corradeFlags
+    MSVC2015_COMPATIBILITY
+    MSVC2017_COMPATIBILITY
+    MSVC2019_COMPATIBILITY
+    BUILD_DEPRECATED
+    BUILD_STATIC
+    BUILD_STATIC_UNIQUE_GLOBALS
+    BUILD_MULTITHREADED
+    TARGET_UNIX
+    TARGET_APPLE
+    TARGET_IOS
+    TARGET_IOS_SIMULATOR
+    TARGET_WINDOWS
+    TARGET_WINDOWS_RT
+    TARGET_EMSCRIPTEN
+    TARGET_ANDROID
+    # TARGET_X86 etc and TARGET_LIBCXX are not exposed to CMake as the meaning
+    # is unclear on platforms with multi-arch binaries or when mixing different
+    # STL implementations. TARGET_GCC etc are figured out via UseCorrade.cmake,
+    # as the compiler can be different when compiling the lib & when using it.
+    PLUGINMANAGER_NO_DYNAMIC_PLUGIN_SUPPORT
+    TESTSUITE_TARGET_XCTEST
+    UTILITY_USE_ANSI_COLORS)
+foreach(_corradeFlag ${_corradeFlags})
+    list(FIND _corradeConfigure "#define CORRADE_${_corradeFlag}" _corrade_${_corradeFlag})
+    if(NOT _corrade_${_corradeFlag} EQUAL -1)
+        set(CORRADE_${_corradeFlag} 1)
+    endif()
+endforeach()
+
+
+# Corrade::rc, a target with just an executable
+if(NOT TARGET Corrade::rc)
+    if(CMAKE_CROSSCOMPILING)
+        find_program(CORRADE_RC_PROGRAM 
+            NAMES corrade-rc 
+            PATHS ENV 
+            PATH NO_DEFAULT_PATH)
+    else()
+        find_program(CORRADE_RC_PROGRAM 
+            NAMES corrade-rc 
+            PATHS "${CMAKE_CURRENT_LIST_DIR}/../../bin/"
+            NO_DEFAULT_PATH)
+    endif()
+
+    get_filename_component(CORRADE_RC_PROGRAM "${CORRADE_RC_PROGRAM}" ABSOLUTE)
+
+    add_executable(Corrade::rc IMPORTED)
+    set_property(TARGET Corrade::rc PROPERTY IMPORTED_LOCATION ${CORRADE_RC_PROGRAM})
```

Comment:
`rc` is built when corrade is built with `with_utility=True`.
Does this logic also work when it is built without rc?

In that case, `CORRADE_RC_PROGRAM` will be `COORADE_RC_PROGRAM-NOT-FOUND`.
Won't cmake complain about this?

Also, perhaps `find_program` should always use the environment variable as hint? `package_info` can set it.

Reply:
I'm testing this. In order to disable utility, you need to disable all the other components: 

```
conan create recipes/corrade/all/conanfile.py corrade/2020.06@ --build=corrade -o corrade:with_utility=False -o corrade:with_testsuite=False -o corrade:with_interconnect=False -o corrade:with_pluginmanager=False
```

...and nothing is left from the library, just three files with some definitions and the `version.h` file 🙈  I'll modify the `test_package` accordingly to support these options. 

BTW, CMake doesn't fail. Anyway, we cannot condition target `Corrade::rc` to `with_utility` option, it can happen that _host_ package is compiled without that executable, but we are in a cross-building scenario and the branch `if (CMAKE_CROSSCOMPILING)` can still find `rc` from the package in the _build_ context.